beacon-chain/sync/validate_aggregate_proof.go

```diff
@@ -101,13 +101,17 @@ func (s *Service) validateAggregatedAtt(ctx context.Context, signed *ethpb.Signe
 	// to weird edge cases during verification. The attestation technically could be used to add value to a block,
 	// but it's invalid in the spirit of the protocol. Here we choose safety over profit.
 	if err := s.chain.VerifyLmdFfgConsistency(ctx, signed.Message.Aggregate); err != nil {
-		fmt.Println(err)
 		traceutil.AnnotateError(span, err)
 		return pubsub.ValidationReject
 	}
 
-	attSlot := signed.Message.Aggregate.Data.Slot
+	// Verify current finalized checkpoint is an ancestor of the block defined by the attestation's beacon block root.
+	if err := s.chain.VerifyFinalizedConsistency(ctx, signed.Message.Aggregate.Data.BeaconBlockRoot); err != nil {
+		traceutil.AnnotateError(span, err)
+		return pubsub.ValidationReject
+	}
 
+	attSlot := signed.Message.Aggregate.Data.Slot
```

Comment:
Can you move `attSlot := signed.Message.Aggregate.Data.Slot` to line 120? (after `AttestationPreState`)